tools/extra/pac/fpgabist/dma/fpga_dma_test.c

```diff
@@ -161,52 +294,95 @@ fpga_result ddr_sweep(fpga_dma_handle dma_h)
    return FPGA_OK;
 }
 
+/* functions to get the bus number when there are multiple buses */
+struct bus_info{
+	uint8_t bus;
+};
+
+fpga_result get_bus_info(fpga_token tok, struct bus_info *finfo){
+	fpga_result res = FPGA_OK;
+	fpga_properties props;
+	res = fpgaGetProperties(tok, &props);
+	ON_ERR_GOTO(res, out, "reading properties from Token");
+
+        fpgaPropertiesGetBus(props, &finfo->bus);
+	ON_ERR_GOTO(res, out_destroy, "Reading bus from properties");
+	
+	out_destroy:
+		fpgaDestroyProperties(&props);
+	out:
+		return res;
+}
+
+void print_bus_info(struct bus_info *info){
+	printf("Running on bus 0x%02X. \n", info->bus);
+}
+
+
 int main(int argc, char *argv[])
 {
    fpga_result res = FPGA_OK;
    fpga_dma_handle dma_h;
    uint64_t count;
-   fpga_properties filter = NULL;
+   int retval = 0;
    fpga_token afc_token;
    fpga_handle afc_h;
    fpga_guid guid;
-   uint32_t num_matches;
    volatile uint64_t *mmio_ptr = NULL;
    uint64_t *dma_buf_ptr  = NULL;
    uint32_t use_ase;
+   uint32_t num_matches = 1;
+   struct bus_info info;
+  
+
 
    if (argc < 2) {
-      printf("Usage: fpga_dma_test <use_ase = 1 (simulation only), 0 (hardware)>");
+      help();
       return 1;
    }
+
+   
+   res = parse_args(argc, argv);
+   if (res < 0){
+	retval = 1;
+	goto out_exit;
+   }
    use_ase = atoi(argv[1]);
-   if (use_ase) {
+   if (use_ase) { 
       printf("Running test in ASE mode\n");
-   } else {
+   } else if (use_ase == 0) {
       printf("Running test in HW mode\n");
    }
+   
 
-   // enumerate the afc
    if (uuid_parse(HELLO_AFU_ID, guid) < 0) {
       return 1;
    }
 
-   res = fpgaGetProperties(NULL, &filter);
-   ON_ERR_GOTO(res, out, "fpgaGetProperties");
-
-   res = fpgaPropertiesSetObjectType(filter, FPGA_ACCELERATOR);
-   ON_ERR_GOTO(res, out_destroy_prop, "fpgaPropertiesSetObjectType");
-
-   res = fpgaPropertiesSetGUID(filter, guid);
-   ON_ERR_GOTO(res, out_destroy_prop, "fpgaPropertiesSetGUID");
-
-   res = fpgaEnumerate(&filter, 1, &afc_token, 1, &num_matches);
-   ON_ERR_GOTO(res, out_destroy_prop, "fpgaEnumerate");
+   res = find_fpga(guid, &afc_token);
```

Comment:
see my prev. comment. res should hold the return code. number of matches should be received in arg.